command/agent.go

```diff
@@ -800,6 +804,7 @@ func (c *AgentCommand) Run(args []string) int {
 				leaseCache.SetShuttingDown(true)
 			}
 			cancelFunc()
+			es.Close()
```

Comment:
Does this need an `if es != nil` or would that be overly defensive?

Reply:
think it would be overly defensive, `es` would only be `nil` if there was an error opening one of the log files and in that case would exit early

or should we check anyway b/c its in a closure?

Reply:
I'll leave it up to you -- I think it wouldn't hurt, but like you said, I can't really see how we'd get to that point